src/components/cc-plan-configurator/cc-plan-configurator.js

```diff
@@ -0,0 +1,199 @@
+    this.value = '';
+
+    /** @type {string}  **/
+    this._currentPlanId = null;
+
+    /** @type {PlanItem[]}  **/
+    this._currentRelatedPlans = [];
+  }
+
+  /**
+   * @param {HTMLInputElementEvent} e
+   * @private
+   */
+  _onChangePlan(e) {
+    const currentPlan = this._findPlan(this.plans, e.target.value);
+    console.log(currentPlan);
+    this._currentPlanId = e.target.value;
+    if (currentPlan.relatedPlans != null && currentPlan.relatedPlans.length > 0) {
+      this.value = currentPlan.relatedPlans[0].id;
+      this._currentRelatedPlans = currentPlan.relatedPlans;
+    } else {
+      this.value = this._currentPlanId;
+      this._currentRelatedPlans = [];
+    }
+
+    dispatchCustomEvent(this, 'input', this.value);
+  }
+
+  /**
+   * @param {HTMLInputElementEvent} e
+   * @private
+   */
+  _onChangeRelatedPlan(e) {
+    this.value = e.target.value;
+    dispatchCustomEvent(this, 'input', this.value);
+  }
+
+  /**
+   * Find a plan by its ID in a list of plans.
+   *
+   * @param {ConfiguratorPlan[]} plans - List of plans to search through
+   * @param {string} planId - ID of the plan to find
+   * @returns {ConfiguratorPlan} The found plan or the first plan in the list if not found
+   * @private
+   */
+  _findPlan(plans, planId) {
+    // We're trying to find if the plan is a related plan or if the plan is a plan with no related plans
+    const exactPlan = plans.find((plan) => {
+      if (plan.relatedPlans == null || plan.relatedPlans.length === 0) {
+        return plan.id === planId;
+      }
+      return plan.relatedPlans.find((relatedPlan) => relatedPlan.id === planId) != null;
+    });
+
+    // If it was the case we return the plan
+    if (exactPlan != null) {
+      return exactPlan;
+    }
+
+    // If it's not one of the case above we try to find the "parent" plan
+    const parentPlan = this.plans.find((plan) => {
+      return plan.id === planId;
+    });
+
+    if (parentPlan != null) {
+      return parentPlan;
+    }
+
+    // If we don't find anything, the id provided wasn't correct so we default to the first plan of the list
+    return plans[0];
+  }
+
+  /**
+   * @param {CcPlanConfiguratorPropertyValues} changedProperties
+   */
+  willUpdate(changedProperties) {
+    if (changedProperties.has('plans') && changedProperties.has('value')) {
+      const currentPlan = this._findPlan(this.plans, this.value);
+      if (currentPlan.relatedPlans != null && currentPlan.relatedPlans.length > 0) {
+        const valueExistsInRelated = currentPlan.relatedPlans.find((plan) => plan.id === this.value) != null;
+        this.value = valueExistsInRelated ? this.value : currentPlan.relatedPlans[0].id;
+      } else {
+        this.value = currentPlan.id;
+      }
+      this._currentPlanId = currentPlan.id;
+      this._currentRelatedPlans = currentPlan.relatedPlans;
+    }
+  }
+
+  render() {
+    return html`
+      <cc-plan-picker
+        @input="${this._onChangePlan}"
```

Comment:
**nitpick:** you're listening to the generic `input` event instead of our custom `cc-plan-picker:input` event. It works because the `input` bubbles but it would be better to listen to the documented event instead